moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -287,31 +294,32 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
 
       if (default_solver_plugin_.empty())
       {
-        ROS_DEBUG_NAMED(LOGNAME, "Loading settings for kinematics solvers from the ROS param server ...");
-
-        // read data using ROS params
-        ros::NodeHandle nh("~");
+        RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER,
+                    "Loading settings for kinematics solvers from the ROS param server ...");
 
+        // TODO (JafarAbdi): Add the appropriate node name for ROS2
+        auto ksolver_params = std::make_shared<rclcpp::SyncParametersClient>(node_, "/dummy_joint_states");
         // read the list of plugin names for possible kinematics solvers
         for (std::size_t i = 0; i < known_groups.size(); ++i)
         {
           std::string base_param_name = known_groups[i].name_;
-          ROS_DEBUG_NAMED("kinematics_plugin_loader", "Looking for param %s ",
-                          (base_param_name + "/kinematics_solver").c_str());
-          std::string ksolver_param_name;
-          bool found = nh.searchParam(base_param_name + "/kinematics_solver", ksolver_param_name);
-          if (!found || !nh.hasParam(ksolver_param_name))
+          std::string ksolver_param_name = base_param_name + ".kinematics_solver";
+          RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "Looking for param %s ", ksolver_param_name.c_str());
+          bool found = ksolver_params->has_parameter(ksolver_param_name);
+          if (!found)
           {
-            base_param_name = robot_description_ + "_kinematics/" + known_groups[i].name_;
-            ROS_DEBUG_NAMED(LOGNAME, "Looking for param %s ", (base_param_name + "/kinematics_solver").c_str());
-            found = nh.searchParam(base_param_name + "/kinematics_solver", ksolver_param_name);
+            base_param_name = robot_description_ + "_kinematics." + known_groups[i].name_;
+            ksolver_param_name = base_param_name + ".kinematics_solver";
+            RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "Looking for param %s ", ksolver_param_name.c_str());
+            found = ksolver_params->has_parameter(ksolver_param_name);
           }
           if (found)
           {
-            ROS_DEBUG_NAMED(LOGNAME, "Found param %s", ksolver_param_name.c_str());
-            std::string ksolver;
-            if (nh.getParam(ksolver_param_name, ksolver))
+            RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "Found param %s", ksolver_param_name.c_str());
+            auto parameters_and_prefixes = ksolver_params->get_parameters({ ksolver_param_name });
```

Comment:
What's your opinion on setting ksolver directly .?
like `std::string ksolver = node_->get_parameter(ksolver_param_name).as_string();` 

when `found` is true `bool Node::get_parameter()` will return true, too - because in both cases the param is declared

Reply:
You're right about the bool. I just had the template function in mind: `node_->get_parameter(ksolver_param_name,  ksolver)`. It seems like both the template function as well as `as_string()` would throw an exception if the type is not string.